chainer/testing/unary_function_test.py

```diff
@@ -0,0 +1,179 @@
+    [-1, 1] is used for both.
+
+    Decorated test class tests forward and backward computation for CPU and GPU
+    across the following :func:`~chainer.testing.parameterize` ed parameters:
+
+    - dtype: ``numpy.float16``, ``numpy.float32`` and ``numpy.float64``
+    - shape: rank of zero and more
+
+    Additionally, it tests the label of Chainer function class if a Chainer
+    function has its corresponding function class. Decorator searches a Chainer
+    function class in ``chainer.functions`` module from name of the Chainer
+    function.
+
+    .. admonition:: Example
+
+       The following code defines a test class that tests trigonometric ``sin``
+       Chainer function that takes a variable with ``dtype`` of float and
+       returns another with the same ``dtype``.
+
+       >>> import unittest, chainer.functions as F
+       >>>
+       >>> @unary_function_test(F.sin)
+       >>> class TestSin(unittest.TestCase):
+       >>>     pass
+
+       Because test methods are implicitly injected to ``TestSin`` class by the
+       decorator, we only place ``pass`` in the class definition.
+
+       We may use this decorator to test unary Chainer functions implemented
+       with composing other Chainer functions, like ``rsqrt`` which computes
+       reciprocal of square root.
+
+       >>> import numpy, unittest, chainer.functions as F
+       >>>
+       >>> def rsqrt(x, dtype=numpy.float32):
+       >>>     return numpy.reciprocal(numpy.sqrt(x, dtype=dtype))
+       >>>
+       >>> @unary_function_test(F.rsqrt, rsqrt)
+       >>> class TestRsqrt(unittest.TestCase):
+       >>>     pass
+
+       Here we define ``rsqrt`` function composing numpy functions to get
+       expected values, passing it to the second argument of
+       ``@unary_function_test`` decorator.
+
+       We may also customize test data to be used. The following is an example
+       of testing ``sqrt`` Chainer function which we want to test in positive
+       value domain leaving some margin around zero of input ``x``.
+
+       >>> def make_data(dtype, shape):
+       >>>     x = numpy.random.uniform(0.1, 1, shape).astype(dtype)
+       >>>     gy = numpy.random.uniform(-1, 1, shape).astype(dtype)
+       >>>     return x, gy
+       >>>
+       >>> @unary_function_test(F.sqrt, make_data=make_data)
+       >>> class TestSqrt(unittest.TestCase):
+       >>>     pass
+
+       We define ``make_data`` function to return input and gradient ndarrays
+       generated in proper value domains with given ``dtype`` and ``shape``
+       parameters, then passing it to the decorator's ``make_data`` argument.
+
+    """
+
+    # Import here to avoid mutual import.
+    from chainer import gradient_check
+    from chainer import testing
+
+    if func_expected is None:
+        name = func.__name__
+        func_expected = getattr(numpy, name)
+
+    if make_data is None:
+        make_data = make_data_default
+
+    def f(klass):
```

Comment:
Please decorate this function with [functools.wraps](https://docs.python.org/2.7/library/functools.html#functools.wraps). Otherwise, `func_name` of this class would be always `f`, regardless of the value of `klass`.